AudiusProject/audius-protocol
# Use nginx ingress instead of sshuttle

### Description

- Uses nginx ingress instead of sshuttle and `audius-compose connect` for aliases
- Refactors docker compose services into multiple files so the root docker-compose.yml is easy to read and comms services can be brought up by passing a new `--comms` flag
- Adds `--libs` flag to `audius-compose up` to not run resource-intensive build-audius-libs container by default

### Tests

- Bring up stack for testing storage v2 with client: `audius-compose up -e 1 --comms` and verify that sign up and uploads work. Note: due to resource constraints it's easier to first do `audius-compose up -e 1` and subsequently do `audius-compose up --comms`
- Bring up default stack (current flow before this PR): `audius-compose up --libs`
- Change a route in nginx_ingress.conf and then do `docker stop ingress && docker compose up -d ingress`
- Ensure you can bring up comms from the `audius-protocol/comms` directory with `make` and do the v2 upload flow

### Monitoring - How will this change be monitored? Are there sufficient logs / alerts?
Only affects local dev. Watch for errors bringing up the stack.
